data-structures/hash-tables.md

```diff
@@ -39,11 +40,31 @@ Knowing how hash tables work will give you a deeper understanding of why hash ta
 
 ### Common Mistakes / Misconceptions
 
-What is the difference between a hash map and a hash table? The two are used interchangeably.
+- What is the difference between a hash map and a hash table? The two are used interchangeably.
+
+- When should I use an array instead of a hash table? If your keys are sequential integers.
+
+- When does a JavaScript object stop being a hash table? When a property is added as a function.
+
+- [hashmap implementation](https://gist.github.com/dtx/3791420)
+
+ ![Image result for difference between hashmap and hashtable](https://cdncontribute.geeksforgeeks.org/wp-content/uploads/hashsetVshashmap.png)
+
+
+### Guided Practice
+
+Let's understand how to make hash tables and hash maps using JavaScript.
+
+- [How to implement simple hash table?](https://medium.freecodecamp.org/how-to-implement-a-simple-hash-table-in-javascript-cb3b9c1f2997)
```

Comment:
The following are great resources, but you also need to specify what you expect apprentices to do with each resource. They have a very limited amount of time to study. Should they read the entire thing, or just a certain part?  Should they watch the whole video? Should they complete the challenge at the end, or skip it?  Are these all required, or are some of them optional?  If they're optional, put them in supplemental materials.  Either way, please describe what they will gain from the resource.